crates/wasmparser/src/validator/types.rs

```diff
@@ -3462,6 +3491,16 @@ where
 }
 
 impl Remapping {
+    /// Add a mapping from the specified old resource ID to the new resource ID
+    pub fn add(&mut self, old: ResourceId, new: ResourceId) {
+        self.resources.insert(old, new);
+    }
+
+    /// Clear any type remappings
+    pub fn reset(&mut self) {
+        self.types.clear()
+    }
```

Comment:
This feels a bit odd in that it only clears out an internal cache as opposed to additionally clearing out the resources mapping. Could this grow a new name to emphasize it's not resetting the entire state but only part of it?